testsuite/tests/typing-modules/firstclass.ml

```diff
@@ -46,6 +46,99 @@ module type S3 = sig type u type t val x : int end
 Line 3, characters 2-67:
 3 |   (x : (module S3 with type t = 'a and type u = 'b) :> (module S'));; (* fail *)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: Type "(module S3 with type t = int and type u = bool)"
-       is not a subtype of "(module S')"
+Error: Type "(module S3 with type t = 'a and type u = 'b)" is not a subtype of
+         "(module S')"
+       The two first-class module types differ by their runtime size.
 |}];;
+
+(* but you cannot move values (no physical coercions) *)
+module type S4 = sig val x : int  val mid:int  val y:int end
+module type S5 = sig val x:int val y:int end
+let g4 x =
+  (x : (module S4) :> (module S5));; (* fail *)
+[%%expect{|
+module type S4 = sig val x : int val mid : int val y : int end
+module type S5 = sig val x : int val y : int end
+Line 4, characters 2-34:
+4 |   (x : (module S4) :> (module S5));; (* fail *)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: Type "(module S4)" is not a subtype of "(module S5)"
+       The two first-class module types do not share
+       the same position for runtime components.
+       For example, the value "y" occurs at the expected position of
+       the value "mid".
```

Comment:
I'm not sure if this is worth fixing, but the message points that `y` in `S5` occurs at the position of `mid` in `S4`, but the references to the package types earlier are in the reverse order, which could be a bit confusing.

Reply:
Keeping the package type and the associated item in the same order is very important. Fixed, thanks !